Index: django/db/backends/mysql/base.py
===================================================================
--- django/db/backends/mysql/base.py	(revision 4485)
+++ django/db/backends/mysql/base.py	(working copy)
@@ -101,6 +101,7 @@
             cursor = self.connection.cursor()
             if self.connection.get_server_info() >= '4.1':
                 cursor.execute("SET NAMES 'utf8'")
+                self.connection.charset = 'utf8'
         else:
             cursor = self.connection.cursor()
         if settings.DEBUG:
