Index: django/middleware/http.py
===================================================================
--- django/middleware/http.py	(revision 4485)
+++ django/middleware/http.py	(working copy)
@@ -48,6 +48,11 @@
     because this sets REMOTE_ADDR based on HTTP_X_FORWARDED_FOR, that means
     anybody can "fake" their IP address. Only use this when you can absolutely
     trust the value of HTTP_X_FORWARDED_FOR.
+
+    By convention, proxy servers augment existing X-Forwarded-For headers by
+    appending their own data to the end. This means that, assuming we're running
+    behind a single trusted proxy, the IP we're looking for is the last one in
+    what might be a comma-separated list of several IPs.
     """
     def process_request(self, request):
         try:
@@ -55,7 +60,5 @@
         except KeyError:
             return None
         else:
-            # HTTP_X_FORWARDED_FOR can be a comma-separated list of IPs.
-            # Take just the first one.
-            real_ip = real_ip.split(",")[0]
+            real_ip = real_ip.split(",")[-1].strip()
             request.META['REMOTE_ADDR'] = real_ip
